packages/app-check/src/internal-api.ts

```diff
@@ -205,6 +205,89 @@ export async function getToken(
   return interopTokenResult;
 }
 
+/**
+ * This function always resolves.
+ * The result will contain an error field if there is any error.
+ * In case there is an error, the token field in the result will be populated with a dummy value
+ */
+export async function getScopedToken(
+  appCheck: AppCheckService
+): Promise<AppCheckTokenResult> {
+  const app = appCheck.app;
+  ensureActivated(app);
+
+  const { provider } = getStateReference(app);
+
+  /**
+   * First check if there is a token in memory from a previous `getToken()` call.
```

Comment:
Some of these comments need to be updated, this used to be a section that checked memory and indexedDB (in getToken()) but it's removed now so it's just variable declarations here.